lib/evaluate.js

```diff
@@ -48,58 +45,52 @@ export const evaluate = async ({
   round.details = sparkRoundDetails
   debug('ROUND DETAILS for round=%s', roundIndex, sparkRoundDetails)
 
-  const started = Date.now()
+  // PERFORMANCE: Avoid duplicating the array of measurements because there are
+  // hundreds of thousands of them. All the function groupMeasurementsToCommittees
+  // needs is to iterate over the accepted measurements once.
+  const iterateAcceptedMeasurements = function * () {
+    for (const m of measurements) {
+      // Mark all measurements as accepted by default.
+      m.taskingEvaluation = 'OK'
+      yield m
+    }
+  }
+
+  const evaluationCommittees = groupMeasurementsToCommittees(iterateAcceptedMeasurements())
+  for (const c of evaluationCommittees.values()) {
+    c.evaluate({ requiredCommitteeSize })
+  }
 
-  const { committees, timings: fraudDetectionTimings } = await runFraudDetection({
+  const committees = Array.from(evaluationCommittees.values())
+  logger.log(
+    'EVALUATE ROUND %s: Evaluated %s measurements.\n',
     roundIndex,
-    measurements,
-    sparkRoundDetails,
-    requiredCommitteeSize,
-    logger
-  })
+    measurements.length
+  )
 
   // Calculate aggregates per evaluation outcome
   /** @type {Partial<Record<import('./typings.js').TaskingEvaluation,  number>>} */
-  const evaluationOutcomes = {
-    OK: 0,
-    TASK_NOT_IN_ROUND: 0,
-    DUP_INET_GROUP: 0,
-    TOO_MANY_TASKS: 0
-  }
-  for (const m of measurements) {
-    evaluationOutcomes[m.taskingEvaluation] = (evaluationOutcomes[m.taskingEvaluation] ?? 0) + 1
-  }
   logger.log(
-    'EVALUATE ROUND %s: Evaluated %s measurements.\n%o',
+    'EVALUATE ROUND %s: Evaluated %s measurements.\n',
     roundIndex,
-    measurements.length,
-    evaluationOutcomes
+    measurements.length
   )
 
-  const fraudDetectionDuration = Date.now() - started
-
   // Telemetry and stats (no rewards)
   recordTelemetry('evaluate', point => {
     point.intField('round_index', roundIndex)
     point.intField('total_measurements', measurements.length)
     point.intField('total_nodes', countUniqueNodes(measurements))
-    point.intField('fraud_detection_duration_ms', fraudDetectionDuration)
-    for (const { influxField, duration } of fraudDetectionTimings) {
-      point.intField(`fraud_detection_timings_${influxField}_ms`, duration)
-    }
-    for (const [type, count] of Object.entries(evaluationOutcomes)) {
-      point.intField(`measurements_${type}`, count)
-    }
```

Comment:
I think this change may break some Grafana charts. I agree that we no longer need to monitor these metrics. After you ship this PR and some rounds are evaluated, please review the Grafana dashboard(s) and remove the charts that visualise these metrics and no longer have data.